base/replutil.jl

```diff
@@ -15,14 +15,14 @@ function show(io::IO, ::MIME"text/plain", iter::Union{KeyIterator,ValueIterator}
         rows < 2 && (print(io, " …"); return)
         cols < 4 && (cols = 4)
         cols -= 2 # For prefix "  "
-        rows -= 2 # For summary and final ⋮ continuation lines
+        rows -= 1 # For summary
     else
         rows = cols = 0
     end
 
     for (i, v) in enumerate(iter)
         print(io, "\n  ")
-        limit && i >= rows && (print(io, "⋮"); break)
+        limit && rows <= i < length(iter) && (print(io, "⋮"); break)
```

Comment:
`length(iter)` is computed only when `rows <= i`, i.e only once for the last iteration: then either `i==length(iter)`, in which case the `enumerate` loop will stop, otherwise `break` is reached. It could be made clearer that this test happens once by writing `rows == i < length(iter)` though.